src/utils.ts

```diff
@@ -27,9 +27,11 @@ export class StreamingClientError extends Error {
 
 export class StreamingSubscriptionError extends Error {
   name = 'StreamingSubscriptionError';
+  missingPermissions?: string[];
 
-  constructor (message: string, public readonly topic?: string, public readonly operation?: 'subscribe' | 'unsubscribe') {
+  constructor (message: string, public readonly topic?: string, public readonly operation?: 'subscribe' | 'unsubscribe', missingPermissions?: string[]) {
```

Comment:
This is fine, but you might consider making this new param an optional object called `details` of which `missingPermissions` is its only field for now. That way we don't have more and more positional params in the future if other bits of detail get added later.